src/client.ts

```diff
@@ -2658,7 +2658,11 @@ export class MatrixClient extends EventEmitter {
         );
     }
 
-    private makeKeyBackupPath(roomId: string, sessionId: string, version: string): IKeyBackupPath {
+    private makeKeyBackupPath(
+        roomId: string | undefined,
+        sessionId: string | undefined,
+        version: string,
+    ): IKeyBackupPath {
```

Comment:
this isn't quite right, `roomId` can only be undefined if `sessionId` is undefined too, this should be an overload signature

Reply:
Hopefully these are all now as expected?

Reply:
Yup, shame that the optional params were put first :(